### PR TITLE
Use new Rizin API instead of cmdRaw/cmdRawAt

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

See https://github.com/rizinorg/rizin/pull/1709 as well. This PR uses some of the Rizin API instead of calling the commands. It needs an updated Rizin to work.

**Test plan (required)**

* edit bytes endian through the stack widget and check it performs it correctly
* double click on stack-widget address or use seekToFunctionStart(TODO: where???) to check that seek and seek history work well
* try to load a file with PDB and make sure PDB info are correctly loaded
* try to edit a variable, both name and type and make sure it works
* test "type link" functionalities to add and remove a type link
* make sure when changing an existing type link the current type name is selected
* check that hex widget can write string
* check that hex widget can write increase/decrease
* check that hex widget can write zeros
* check that hex widget can write base64 encoded/decoded

* add CutterRzListForeach as a foreach-keyword for clang-format

**Closing issues**

Address parts of #2666

### CODE REVIEW

File: src/core/Cutter.cpp [hunk 1]
```diff
@@ -1918,7 +1925,7 @@ void CutterCore::attachRemote(const QString &uri)
         RzList *descs = rz_id_storage_list(core->io->files);
         RzListIter *it;
         RzIODesc *desc;
-        CutterRzListForeach(descs, it, RzIODesc, desc) {
+        CutterRzListForeach (descs, it, RzIODesc, desc) {
```
Comment (xarkes) [inline src/core/Cutter.cpp:1928]: Is there any reason for this extra space? Is it how you use the API in Rizin, or is it clang-format doing it?

File: src/dialogs/EditVariablesDialog.cpp [hunk 2]
```diff
@@ -55,7 +55,25 @@ void EditVariablesDialog::applyFields()
...
+    if (!fcn) {
+        return;
+    }
+
+    RzAnalysisVar *v = rz_analysis_function_get_var_byname(fcn, desc.name.toUtf8().constData());
+    if (!v) {
+        return;
+    }
+
+    char *error_msg = NULL;
```
Comment (XVilka) [inline src/dialogs/EditVariablesDialog.cpp:69]: Maybe we should have a shortcut like `rz_anaysis_var_set_type_str(v, type_str)` in the Rizin API. But it's fine as is already.

**Review by XVilka — APPROVED** [review]

**Comment by ret2libc** [comment]
> make sure when changing an existing type link the current type name is selected

Seems to fail

**Comment by ret2libc** [comment]
> check that hex widget can write increase/decrease

There's a problem if you change from byte to something else in the dialog, but this is present also in last release.

**Comment by ret2libc** [comment]
> check that hex widget can write base64 encoded/decoded

Also this fails.

**Comment by ret2libc** [comment]
> > make sure when changing an existing type link the current type name is selected
> 
> Seems to fail

@thestr4ng3r @xarkes i'm not sure about this... it seems that sometimes it work, others it doesn't :/ maybe it depends on where exactly you put the cursor? Can you double check? I was not able to use `type link` functions in last released cutter so i could not really compare.

**Review by xarkes — CHANGES_REQUESTED** [review]
- [ ] edit bytes endian through the stack widget and check it performs it correctly
    * couldn't find this option during debug
- [x] double click on stack-widget address or use seekToFunctionStart(TODO: where???) to check that seek and seek history work well
    * seek works well, history works well (although ESC key seems not to be working)
- [ ] try to load a file with PDB and make sure PDB info are correctly loaded
    * no PDB around, didn't test
- [x] try to edit a variable, both name and type and make sure it works
- [ ] test "type link" functionalities to add and remove a type link
    * it seems to be working in order to set a type link however I couldn't find how to remove it
- [ ] make sure when changing an existing type link the current type name is selected
    * there are weird issues with the pre-fill of the dialog box but I'm not sure it's unrelated to this PR
- [x] check that hex widget can write string
    * tested with cache mode
- [ ] check that hex widget can write increase/decrease
    * I couldn't understand the UI, it asks for a value (input is empty) and I'm not sure what to do with it
- [x] check that hex widget can write zeros
- [x] check that hex widget can write base64 encoded/decoded

**Comment by thestr4ng3r** [comment]
macOS/arm64 with bundled rizin:

* [ ] ~edit bytes endian through the stack widget and check it performs it correctly~ **can't test on macOS atm because debugging there is too broken, but that is nothing new.**
* [ ] ~double click on stack-widget address or use seekToFunctionStart(TODO: where???) to check that seek and seek history work well~ **same**
* [x] try to load a file with PDB and make sure PDB info are correctly loaded - **works both with the menu entry and from InitialOptionsDialog**
* [x] try to edit a variable, both name and type and make sure it works
* [ ] test "type link" functionalities to add and remove a type link - **seems entirely broken and breaks disassembly, I suggest to remove it from cutter in light of globals which can be added later**
* [ ] make sure when changing an existing type link the current type name is selected - **same**
* [x] check that hex widget can write string
* [x] check that hex widget can write increase/decrease
* [x] check that hex widget can write zeros
* [x] check that hex widget can write base64 encoded/decoded

**Review by thestr4ng3r — APPROVED** [review]
So looks like apart from type links, everything works. But type links are broken primarily from rizin side, so I would merge this and then remove them from cutter.

**Comment by ret2libc** [comment]
> edit bytes endian through the stack widget and check it performs it correctly

this is actually just setting a value in the stack widget, sorry

**Comment by XVilka** [comment]
I suggest the following:
- We remove typelinks in 2.0.3 release
- We add global variables support in shortly following 2.0.4 release
- We remove typelinks from Rizin in maybe 0.3.1

**Pull request merged**